local/src/core/http.cc

```diff
@@ -1227,8 +1231,81 @@ Session::set_fd(int fd)
   return errata;
 }
 
+InterfaceNameToEndpoint::InterfaceNameToEndpoint(TextView expected_interface, int expected_family)
+  : _expected_interface{expected_interface}
+  , _expected_family{expected_family}
+{
+}
+
+InterfaceNameToEndpoint::~InterfaceNameToEndpoint()
+{
+  if (_ifaddr_list_head != nullptr) {
+    freeifaddrs(_ifaddr_list_head);
+    _ifaddr_list_head = nullptr;
+  }
+}
+
+swoc::Rv<struct ifaddrs *>
+InterfaceNameToEndpoint::find_matching_interface()
+{
+  swoc::Rv<struct ifaddrs *> zret{nullptr};
+  if (_ifaddr_list_head == nullptr) {
+    if (getifaddrs(&_ifaddr_list_head) == -1) {
+      zret.error("getifaddrs failed: {}", swoc::bwf::Errno{});
+      return zret;
+    }
+  }
+  for (auto *ifa = _ifaddr_list_head; ifa != nullptr; ifa = ifa->ifa_next) {
+    std::string_view interface_name{ifa->ifa_name};
+    if (interface_name != _expected_interface) {
+      continue;
+    }
+    auto const family = ifa->ifa_addr->sa_family;
+    if (family != _expected_family) {
+      continue;
+    }
+    zret = ifa;
+    return zret;
+  }
+  return nullptr;
+}
+
+swoc::Rv<swoc::IPEndpoint>
+InterfaceNameToEndpoint::find_ip_endpoint()
+{
+  swoc::Rv<swoc::IPEndpoint> zret{};
+  if (getifaddrs(&_ifaddr_list_head) == -1) {
+    zret.error("getifaddrs failed: {}", swoc::bwf::Errno{});
+    return zret;
+  }
+  auto &&[matching_interface, find_errata] = find_matching_interface();
+  zret.note(std::move(find_errata));
+  if (!zret.is_ok()) {
+    return zret;
+  }
+  if (matching_interface == nullptr) {
+    zret.error(
+        "Could not find an interface named {} with family {}.",
+        _expected_interface,
+        swoc::IPEndpoint::family_name(_expected_family));
+    return zret;
+  }
+  auto &ip_endpoint = zret.result();
+  ip_endpoint.assign(matching_interface->ifa_addr);
+  std::string ip_string;
+
+  // Convert to string format
+  swoc::bwprint(ip_string, "{}", ip_endpoint);
+  zret.diag(
+      "Found interface from name {} with family {} and ip {}",
+      _expected_interface,
+      swoc::IPEndpoint::family_name(ip_endpoint.family()),
+      ip_string);
```

Comment:
Oh wait, maybe we don't need ip_string at all. @willwendorfvm : can you try just passing int ip_endpoint for this argument?